File: Difficulty/Easy/[169] Majority Element/solution.cpp
/*
 * @lc app=leetcode id=169 lang=cpp
 *
 * [169] Majority Element
 */

// @lc code=start
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int half_len = nums.size() / 2;
        unordered_map<int, int> map;
        for (auto& num : nums) {
            map[num]++;
            if (map[num] > half_len)
                return num;
        }
        return 0;
    }
};
// @lc code=end

File: Difficulty/Easy/[169] Majority Element/solution.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        half_len, m = len(nums) // 2, dict()
        for num in nums:
            if m.get(num, 0):
                m[num] += 1
            else:
                m[num] = 1
            if m[num] > half_len:
                return num